platform-sdk/swirlds-platform-core/src/main/java/com/swirlds/platform/state/State.java

```diff
@@ -155,13 +151,20 @@ public PlatformState getPlatformState() {
     }
 
     /**
-     * Set the platform state.
+     * Updates the platform state.
      *
-     * @param platformState the platform state
+     * @param platformStateAccessor the platform state
      */
     @Override
-    public void setPlatformState(@NonNull final PlatformState platformState) {
-        setChild(ChildIndices.PLATFORM_STATE, platformState);
+    public void updatePlatformState(@NonNull final PlatformStateAccessor platformStateAccessor) {
+        if (platformStateAccessor instanceof PlatformState platformState) {
```

Comment:
probably obvious but i fail to understand why we need this check. PlatformStateAccessor has 3 impls - PlatformState, R/WStateStores. Are we throwing an exception if it's one of R/WStateStores? The tests i could find all use `PlatformState` so not sure.

Reply:
The point of this check is that this class is not supposed to be used with the other implementation because `State` class is not used in Hedera app. However, we still need this class for demo apps.